### PR TITLE
ref(ui): converted sidebarMenuItem to ts

### PR DESCRIPTION
**Pull request by priscilawebdev**
Converted the SidebarMenuItem component from JSX to Typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/sidebar/sidebarMenuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  );
+};
+
+export default SidebarMenuItem;
+
+export const menuItemStyles = (p: SidebarMenuItemLinkProps & {theme: Theme}) => css`
+  color: ${p.theme.gray5};
+  cursor: pointer;
+  display: flex;
+  font-size: 14px;
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/sidebar/sidebarMenuItem.tsx]:
```suggestion
  font-size: ${p => p.theme.fontSizeMedium};
```

File: src/sentry/static/sentry/app/components/sidebar/sidebarMenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const SidebarMenuItem = ({to, href, children, ...props}: Props) => {
+  const hasMenu = !to && !href;
+  return (
+    <StyledSidebarMenuItemLink to={to} href={href} {...props}>
+      <MenuItemLabel hasMenu={hasMenu}>{children}</MenuItemLabel>
+    </StyledSidebarMenuItemLink>
+  );
+};
+
+export default SidebarMenuItem;
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/sidebar/sidebarMenuItem.tsx]: Should this be moved to the bottom of the file?

File: src/sentry/static/sentry/app/components/sidebar/sidebarMenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+export const menuItemStyles = (p: SidebarMenuItemLinkProps & {theme: Theme}) => css`
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/sidebar/sidebarMenuItem.tsx]: Maybe we could not export it directly, but do a named export at the bottom of the file next to default one?

File: src/sentry/static/sentry/app/components/sidebar/help.jsx [hunk 4]
```diff
@@ -51,10 +51,10 @@ class SidebarHelp extends React.Component {
                 <SidebarMenuItem onClick={this.handleSearchClick}>
                   {t('Search Docs and FAQs')}
                 </SidebarMenuItem>
-                <SidebarMenuItem href="https://forum.sentry.io/" target="_blank">
+                <SidebarMenuItem href="https://forum.sentry.io/" openInANewTab>
                   {t('Community Discussions')}
                 </SidebarMenuItem>
-                <SidebarMenuItem href="https://status.sentry.io/" target="_blank">
+                <SidebarMenuItem href="https://status.sentry.io/" openInANewTab>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/sidebar/help.jsx]: `inNewTab` is a shorter way to say this. The `InANew` part looks awkward to me. :bike::house:

**Review by matejminar — APPROVED** [review]

**Pull request merged**